pkg/services/store/kind/dashboard/summary.go

```diff
@@ -0,0 +1,106 @@
+package dashboard
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strconv"
+	"sync"
+
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/services/sqlstore"
+	"github.com/grafana/grafana/pkg/services/store"
+)
+
+func GetObjectKindInfo() models.ObjectKindInfo {
+	return models.ObjectKindInfo{
+		ID:          models.StandardKindDashboard,
+		Name:        "Dashboard",
+		Description: "Define a grafana dashboard layout",
+	}
+}
+
+func NewDashboardSummary(sql *sqlstore.SQLStore) models.ObjectSummaryBuilder {
+	cache := make(map[int64]models.ObjectSummaryBuilder)
```

Comment:
How are we refreshing this cache? What if someone adds a datasource while grafana is running?

Reply:
updated so we do it on every request -- not efficient, but fine for now